src/map.rs

```diff
@@ -160,15 +160,35 @@ where
     K: Sync + Send + Clone + Hash + Eq,
     V: Sync + Send,
 {
-    /// Creates a new, empty map with the default initial table size (16).
+    /// Creates an empty `HashMap`.
+    ///
+    /// The hash map is initially created with a capacity of 0, so it will not allocate until it
+    /// is first inserted into.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use flurry::HashMap;
+    /// let map: HashMap<&str, i32> = HashMap::new();
+    /// ```
     pub fn new() -> Self {
         Self::default()
     }
 
-    /// Creates a new, empty map with an initial table size accommodating the specified number of
-    /// elements without the need to dynamically resize.
-    pub fn with_capacity(n: usize) -> Self {
-        Self::with_capacity_and_hasher(n, crate::DefaultHashBuilder::default())
+    /// If `capacity` is 0, the hash map will not allocate
+    /// Creates an empty `HashMap` with the specified capacity.
+    ///
+    /// The hash map will be able to hold at least `capacity` elements without
+    /// reallocating. If `capacity` is 0, the hash map will not allocate.
+    ///
```

Comment:
We may want to mention here that our `with_capacity` (as opposed to `std::collections::HashMap` I believe) does not _guarantee_ that we won't resize if `capacity` elements are inserted. In general, the map will be resized once a certain amount of collisions are detected, so with particularly bad key distributions, we may resize sooner.